x/pairing/keeper/filters/frozen_providers_filter.go

```diff
@@ -0,0 +1,28 @@
+package filters
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	projectstypes "github.com/lavanet/lava/x/projects/types"
+)
+
+type FrozenProvidersFilter struct{}
+
+func (f *FrozenProvidersFilter) Filter(ctx sdk.Context, providers []epochstoragetypes.StakeEntry) []bool {
+	filterResult := make([]bool, len(providers))
+	for i := range providers {
+		if !isProviderFrozen(ctx, providers[i]) {
+			filterResult[i] = true
+		}
+	}
+
+	return filterResult
+}
+
+func (f *FrozenProvidersFilter) InitFilter(strictestPolicy projectstypes.Policy) bool {
+	return true
+}
+
+func isProviderFrozen(ctx sdk.Context, stakeEntry epochstoragetypes.StakeEntry) bool {
+	return stakeEntry.StakeAppliedBlock > uint64(ctx.BlockHeight())
```

Comment:
return stakeEntry.StakeAppliedBlock == FROZEN_PROVIDER(we have a const for that)

Reply:
Can't find the const

Reply:
msg_server_freeze.go line 14

Reply:
done